src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -2658,19 +2658,26 @@ private void EnsureTypeParametersAreLoaded()
             {
                 if (_lazyTypeParameters.IsDefault)
                 {
+                    if (_arity == 0)
+                    {
+                        _lazyTypeParameters = ImmutableArray<TypeParameterSymbol>.Empty;
+                        return;
+                    }
+
                     var moduleSymbol = ContainingPEModule;
 
                     // If this is a nested type generic parameters in metadata include generic parameters of the outer types.
                     int firstIndex = _genericParameterHandles.Count - _arity;
 
-                    TypeParameterSymbol[] ownedParams = new TypeParameterSymbol[_arity];
```

Comment:
@333fred avoiding this unnecessary allocation on every type does seem like a nice to have.

Reply:
Huh?